test.js

```diff
@@ -1,17 +1,49 @@
 import test from 'ava';
 import sortKeys from '.';
 
+function deepEqualInOrder(t, actual, expected) {
+	t.deepEqual(actual, expected);
+
+	const seen = new Set();
+
+	function assertSameKeysInOrder(object1, object2) {
+		// This function assumes the objects given are already deep equal.
+
+		if (seen.has(object1) && seen.has(object2)) {
+			return;
+		}
+
+		seen.add(object1);
+		seen.add(object2);
+
+		if (Array.isArray(object1)) {
+			for (let i = 0; i < object1.length; i++) {
+				assertSameKeysInOrder(object1[i], object2[i]);
+			}
+		} else if (typeof object1 === 'object') {
+			const keys1 = Object.keys(object1);
+			const keys2 = Object.keys(object2);
+			t.deepEqual(keys1, keys2);
+			for (let i = 0; i < keys1.length; i++) {
+				assertSameKeysInOrder(object1[keys1[i]], object2[keys2[i]]);
+			}
+		}
+	}
+
+	assertSameKeysInOrder(actual, expected);
+}
+
 test('sort the keys of an object', t => {
-	t.deepEqual(sortKeys({c: 0, a: 0, b: 0}), {a: 0, b: 0, c: 0});
+	deepEqualInOrder(t, sortKeys({c: 0, a: 0, b: 0}), {a: 0, b: 0, c: 0});
 });
 
 test('custom compare function', t => {
-	const compare = (a, b) => a.localeCompare(b);
-	t.deepEqual(sortKeys({c: 0, a: 0, b: 0}, {compare}), {c: 0, b: 0, a: 0});
+	const compare = (a, b) => b.localeCompare(a);
+	deepEqualInOrder(t, sortKeys({c: 0, a: 0, b: 0}, {compare}), {c: 0, b: 0, a: 0});
```

Comment:
After introducing `deepEqualInOrder`, it turned out this test was actually incorrect.

Reply:
Nice catch ❤️